aws/resource_aws_backup_plan_test.go

```diff
@@ -0,0 +1,184 @@
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "version"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsBackupPlan_withRuleRemove(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsBackupPlanDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBackupPlanWithRules(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsBackupPlanExists("aws_backup_plan.test"),
+					resource.TestCheckResourceAttr("aws_backup_plan.test", "rule.#", "2"),
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "arn"),
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "version"),
+				),
+			},
+			{
+				Config: testAccBackupPlanConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsBackupPlanExists("aws_backup_plan.test"),
+					resource.TestCheckResourceAttr("aws_backup_plan.test", "rule.#", "1"),
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "arn"),
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "version"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsBackupPlan_withRuleAdd(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsBackupPlanDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBackupPlanConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsBackupPlanExists("aws_backup_plan.test"),
+					resource.TestCheckResourceAttr("aws_backup_plan.test", "rule.#", "1"),
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "arn"),
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "version"),
+				),
+			},
+			{
+				Config: testAccBackupPlanWithRules(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsBackupPlanExists("aws_backup_plan.test"),
+					resource.TestCheckResourceAttr("aws_backup_plan.test", "rule.#", "2"),
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "arn"),
+					resource.TestCheckResourceAttrSet("aws_backup_plan.test", "version"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsBackupPlanDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).backupconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_backup_plan" {
+			continue
+		}
+
+		input := &backup.GetBackupPlanInput{
+			BackupPlanId: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.GetBackupPlan(input)
+
+		if err == nil {
+			if *resp.BackupPlanId == rs.Primary.ID {
+				return fmt.Errorf("Plane '%s' was not deleted properly", rs.Primary.ID)
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsBackupPlanExists(name string) resource.TestCheckFunc {
```

Comment:
The `Exists` functions in Terraform Provider acceptance testing are designed to check the remote API to ensure the requested resource was actually created (and preferably return a SDK object for other potential acceptance testing usage like we need to create a `Disappears` function). More information can be found in the [Extending Terraform documentation](https://www.terraform.io/docs/extend/testing/acceptance-tests/teststep.html#custom-check-functions).

```go
func testAccCheckAwsBackupPlanExists(resourceName string, backupPlan *backup.GetBackupPlanOutput) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		rs, ok := s.RootModule().Resources[resourceName]

		if !ok {
			return fmt.Errorf("Not found: %s", resourceName)
		}

		if rs.Primary.ID == "" {
			return fmt.Errorf("Resource ID is not set")
		}

		conn := testAccProvider.Meta().(*AWSClient).backupconn

		input := &backup.GetBackupPlanInput{
			BackupPlanId: aws.String(rs.Primary.ID),
		}

		output, err := conn.GetBackupPlan(input)

		if err != nil {
			return err
		}

		*backupPlan = *output

		return nil
	}
}
```